src/App.tsx

```diff
@@ -1,26 +1,289 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { createTodos, getTodos, updateTodos, deleteTodos } from './api/todos';
 
-const USER_ID = 0;
+enum Filter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+
+  const [counterOfActiveTodos, setCounterOfActiveTodos] = useState(0);
```

Comment:
it can be the simple var
```suggestion
  const counterOfActiveTodos = todos.filter(todo => !todo.completed).length;
```